### DISCUSSION TITLE
Dataclass style SQLAlchemy models with lazy relationship fail in _prepare_response_content

### DISCUSSION THREAD

**Discussion by solarmist** [fastapi/fastapi#8824] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from dataclasses import dataclass
from fastapi import APIRouter
from sqlalchemy import Integer
from sqlalchemy.orm import registry

router = APIRouter()
mapper_registry = registry()

@mapper_registry.mapped
@dataclass
class User:
    __tablename__ = "user"
    __sa_dataclass_metadata_key__ = "sa"

    id: int = field(
        metadata={
            "sa": Column(
                Integer,
                primary_key=True,
                default=uuid4,
                nullable=False,
            )
        },
    )
    email: str = field(metadata={"sa": Column(String, unique=True, nullable=False)})
    state: "UserState" = field(
        # These are explicitly init=True because they should always exist with the
        # user record
        init=True,
        metadata={
            "sa": relationship(
                "UserState",
                uselist=False,
            )
        },
    )


@mapper_registry.mapped
@dataclass
class UserState:
    """The user state table.

    Information about the user and their use of the app.
    """

    __tablename__ = "user_state"
    __sa_dataclass_metadata_key__ = "sa"

    timezone: str = field(
        metadata={
            "sa": Column(
                String,
                nullable=False,
            )
        },
    )

@router.post("/login/test-token", response_model=schemas.User)
def test_token(
    current_user: models.User = Depends(deps.get_current_user),
) -> Any:
 
    # The DB session is closed then current_user is mapped to the response_model, but during that
    # current_user.history is accessed and fails.
    return current_user
```

### Description

For dataclass style SQLAlchemy models which have lazy relationships, there is no check to ensure "from_orm" is used.

https://github.com/tiangolo/fastapi/blob/bee35f5ae1fc58e7ab125427ad4287210e99d8b3/fastapi/routing.py#L102

### Operating System

macOS

### Operating System Details

11.4

### FastAPI Version

0.68.1

### Python Version

3.9

**Comment by sandys** [comment]
@solarmist did u find a solution to this issue ?

**Comment by solarmist** [comment]
Just to access lazy loaded data before returning.
